cirq-core/cirq/study/result.py

```diff
@@ -201,13 +202,20 @@ def multi_measurement_histogram(
             A counter indicating how often measurements sampled various
             results.
         """
-        fixed_keys = tuple(_key_to_str(key) for key in keys)
-        samples: Iterable[Any] = zip(*(self.measurements[sub_key] for sub_key in fixed_keys))
-        if len(fixed_keys) == 0:
-            samples = [()] * self.repetitions
+        fixed_keys = [_key_to_str(key) for key in keys]
         c: collections.Counter = collections.Counter()
-        for sample in samples:
-            c[fold_func(sample)] += 1
+        if len(fixed_keys) == 0:
+            c[None] += self.repetitions
+            return c
+        bit_counts = [self.measurements[key].shape[1] for key in fixed_keys]
```

Comment:
For what it's worth, this is a bit of a hack, but I don't actually think it's possible to reliably extract the number of qubits that participated in a measurement from result.data.